mne_neuron/dipole.py

```diff
@@ -14,55 +14,109 @@ def _hammfilt(x, winsz):
     return convolve(x, win, 'same')
 
 
-def simulate_dipole(net, n_jobs=1):
-    """Simulate a dipole given the experiment parameters.
+def rmse (a1, a2):
+    from numpy import sqrt
+
+    # return root mean squared error between a1, a2; assumes same lengths, sampling rates
+    len1,len2 = len(a1),len(a2)
+    sz = min(len1,len2)
+    return sqrt(((a1[0:sz] - a2[0:sz]) ** 2).mean())
+
+
+def calcerr (ddat):
+    from scipy import signal
+
+    # calculates RMSE error from ddat
+    # first downsample simulation timeseries to 600 Hz (assumes same time length as data)
+    dpldown = signal.resample(ddat['dpl']['agg'], len(ddat['dextdata']))
+    err0 = rmse(ddat['dextdata'][:,1], dpldown)
+    return err0
+
+
+def initialize_sim_once(net):
+    """
+    Initialize NEURON simulation variables
 
     Parameters
     ----------
     net : Network object
-        The Network object specifying how cells are
-        connected.
-    n_jobs : int
-        The number of jobs to run in parallel
-
-    Returns
-    -------
-    dpl: instance of Dipole
-        The dipole object
+        The Network object with parameter values
     """
-    from .parallel import rank, nhosts, pc, cvode
-    from .network import Network
 
+    from .parallel import pc
     from neuron import h
     h.load_file("stdrun.hoc")
 
-    net = Network(net.params, n_jobs=n_jobs)
+    global t_vec, dp_rec_L2, dp_rec_L5
 
-    # Now let's simulate the dipole
-    if rank == 0:
-        print("running on %d cores" % nhosts)
+    t_vec = h.Vector()
+    dp_rec_L2 = h.Vector()
+    dp_rec_L5 = h.Vector()
 
     # global variables, should be node-independent
     h("dp_total_L2 = 0.")
     h("dp_total_L5 = 0.")
 
+    t_vec.record(h._ref_t)  # time recording
+    dp_rec_L2.record(h._ref_dp_total_L2)  # L2 dipole recording
+    dp_rec_L5.record(h._ref_dp_total_L5)  # L5 dipole recording
+
     # Set tstop before instantiating any classes
     h.tstop = net.params['tstop']
     h.dt = net.params['dt']  # simulation duration and time-step
     h.celsius = net.params['celsius']  # 37.0 - set temperature
 
-    # We define the arrays (Vector in numpy) for recording the signals
-    t_vec = h.Vector()
-    t_vec.record(h._ref_t)  # time recording
-    dp_rec_L2 = h.Vector()
-    dp_rec_L2.record(h._ref_dp_total_L2)  # L2 dipole recording
-    dp_rec_L5 = h.Vector()
-    dp_rec_L5.record(h._ref_dp_total_L5)  # L5 dipole recording
 
-    net.movecellstopos()  # position cells in 2D grid
+def simulate_dipole(net, trial=0, inc_evinput=0.0, print_progress=True, extdata=None):
+    """Simulate a dipole given the experiment parameters.
 
-    # sets the default max solver step in ms (purposefully large)
-    pc.set_maxstep(10)
+    Parameters
+    ----------
+    net : Network object
+        The Network object specifying how cells are
+        connected.
```

Comment:
nitpick, but the numpy docstring conventions suggests that there should not be any space between consecutive parameters in the docstring